pkg/api/registry.proto

```diff
@@ -37,6 +38,14 @@ message GroupVersionKind{
 	string plural = 4;
 }
 
+message Dependency{
```

Comment:
Since we will have multiple sets of fields based on the type of the dependency, we should do one of the following:

1. Have one type field, and another "value" field, that we parse in a second step, i.e.:

```proto
message Dependency{
  string type = 1;
  string value = 2;
}
```
```go
if dep.type == "olm.gvk" {
   gvktype = parseGVKValue(dep.value)
}
```

2. Use a union type to support future field variants, i.e.:

```proto
message Dependency {
  string type = 1;
  oneof deptype {
     GVKDependency gvk = 2;
     PackageDependency pkg = 3;
  }
}

message GVKDependency {
  string group = 1;
  string version = 2;
  string kind = 3;
}

message PackageDependency {
  string name = 1;
  string version = 2;
}
```